### PR TITLE
Added support for ADS1015

### PR DESCRIPTION
**Pull request by xvil**
# What does this implement/fix?

Added support for ADS1015 (12-bits flavour of ADS1115)

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2581

## Test Environment

- [ ] ESP32
- [X] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
# Assuming i2c has been correctly set up

ads1115:
   # Slave address of the ADS1015 on I2C bus
  - address: 0x48 #or whatever slave address the ADS1015 in available on (i2c scan can be used to discover it)

sensor:
  - platform: ads1115
    id: amb_temp_voltage
    internal: true
    multiplexer: 'A0_GND'
    gain: 4.096
    resolution: 12 bits # could also be "12_BITS" or "12 BITS"
    update_interval: 1000ms
    name: "Ambient NTC voltage"
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/ads1115/ads1115.cpp [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #include "ads1115.h"
 #include "esphome/core/log.h"
 #include "esphome/core/hal.h"
+#include <sstream>
+#include <iomanip>
```
Comment (jesserockz) [inline esphome/components/ads1115/ads1115.cpp]: Why were these added?
```suggestion
```

Comment (xvil) [reply esphome/components/ads1115/ads1115.cpp]: It was added for some additional debugging code that I removed.
I have removed those unneeded dependencies now.

**Review by jesserockz — APPROVED** [review]
